src/wallet/wallet_tx_builder.h

```diff
@@ -185,6 +193,17 @@ class TransactionEffects {
         return spendable;
     }
 
+    /**
+     * This is automatically called by the constructor, so it’s not generally necessary to call this
+     * otherwise.
+     */
+    void LockSpendable() const;
+
+    /**
+     * This should be called just before the `TransactionEffects` goes out of scope.
```

Comment:
I'm not sure about this comment; it's not really about when it goes out of scope. If we make a 2-phase, propose-and-then-approve API available, we'll want the unlock to happen either if attempting to create the transaction results in an error, or when the resulting transaction is actually confirmed on-chain to some depth. 

I think that the implementation is essentially fine for now, but the comment should take into account this kind of future use.

Reply:
Though, this does make me wonder how locking as it already exists interacts with runtime failures that cause the node to crash while notes are locked. Is this "locking" exclusively in-memory?

Reply:
Yeah, locking is in-memory, so crashing “unlocks” everything.

I think anything that needs the locks to persist longer would want to move this implementation out of `TransactionEffects` anyway, as this is restricted to the life of that object. This comment is more about avoiding early unlocking and/or carrying around an “invalid” `TransactionEffects`.

Reply:
Should this comment now be worded differently?

Reply:
Reworded.